### PR TITLE
Break the cactus block when it grows next to a block.

### PR DESCRIPTION
**Pull request by Bond-009**
Stops this from happening:
screen shot 2017-07-13 at 16 26 30

### CODE REVIEW

File: src/Chunk.cpp [hunk 1]
```diff
@@ -1142,11 +1142,18 @@ int cChunk::GrowCactus(int a_RelX, int a_RelY, int a_RelZ, int a_NumBlocks)
...
-		if (UnboundedRelGetBlock(a_RelX, Top + i, a_RelZ, BlockType, BlockMeta) && (BlockType == E_BLOCK_AIR))
+		if (UnboundedRelGetBlockType(a_RelX, Top + i, a_RelZ, BlockType) && (BlockType == E_BLOCK_AIR))
 		{
-			// TODO: Check the surrounding blocks, if they aren't air, break the cactus block into pickups (and continue breaking blocks above in the next loop iterations)
 			UnboundedRelFastSetBlock(a_RelX, Top + i, a_RelZ, E_BLOCK_CACTUS, 0);
+			if (
+				(UnboundedRelGetBlockType(a_RelX + 1, Top + i, a_RelZ, BlockType) && (BlockType == E_BLOCK_AIR)) ||
+				(UnboundedRelGetBlockType(a_RelX - 1, Top + i, a_RelZ, BlockType) && (BlockType == E_BLOCK_AIR)) ||
+				(UnboundedRelGetBlockType(a_RelX, Top + i, a_RelZ + 1, BlockType) && (BlockType == E_BLOCK_AIR)) ||
+				(UnboundedRelGetBlockType(a_RelX, Top + i, a_RelZ - 1, BlockType) && (BlockType == E_BLOCK_AIR))
```
Comment (madmaxoft) [inline src/Chunk.cpp]: The wiki says "solid material", so things such as torches should be acceptable, too.

Comment (Bond-009) [inline src/Chunk.cpp]: 👍 Fixed

Comment (madmaxoft) [inline src/Chunk.cpp]: Oh, I thought this should first be researched, the wiki is way too vague in its wording, I wasn't sure.

Comment (Bond-009) [inline src/Chunk.cpp]: I checked it in Vanilla first, non-solids like torches and redstone comparators can be placed next to cacti, but solid blocks like fences etc cant

**Pull request merged**